### PR TITLE
Reader Recent Feed Overhaul: Match cog menu styles and update posts list styles

### PR DESCRIPTION
**Pull request by allilevine**
Related to https://github.com/Automattic/loop/issues/222

## Proposed Changes

* Import base and DataViews styles to match the cog menu styles.
* Update the posts list styles to incorporate the imported styles.
* Use the site icon for the list media field in DataViews.

Before | After
---- | ----
Screen Shot 2024-11-25 at 4 40 18 PM | Screen Shot 2024-11-25 at 4 40 12 PM
Screen Shot 2024-11-25 at 4 40 49 PM | Screen Shot 2024-11-25 at 4 40 31 PM

## Why are these changes being made?

* pe7F0s-2iv-p2

## Testing Instructions

* Go to /read?flags=reader/recent-feed-overhaul
* Click on the cog icon (next to Search, under Recent).
* The styles should match those in /sites and https://wordpress.github.io/gutenberg/?path=/docs/dataviews-dataviews--docs
* Check that the posts list looks good at different screen sizes. Expected differences with staging: 
  * The loading animation is centered
  * Selected items have blue text
  * Items are at least 52px tall

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/reader/recent/index.tsx [hunk 1]
```diff
@@ -75,10 +79,12 @@ const Recent = () => {
 	const fields = useMemo(
 		() => [
 			{
-				id: 'seen',
-				label: translate( 'Seen' ),
+				id: 'icon',
+				label: translate( 'Icon' ),
 				render: ( { item }: { item: ReaderPost } ) => {
-					return <RecentSeenField post={ getPostFromItem( item ) } />;
+					const post = getPostFromItem( item );
```
Comment (allilevine) [inline client/reader/recent/index.tsx:85]: I removed the seen field and replaced it with an icon field that renders the icon in the DataViews `mediaField`.

File: client/reader/recent/style.scss [hunk 2]
```diff
@@ -1,4 +1,6 @@
-@import '@wordpress/base-styles/breakpoints';
+@import "@wordpress/base-styles/breakpoints";
+@import "@wordpress/base-styles/mixins";
+@import "@wordpress/dataviews/build-style/style.css";
```
Comment (allilevine) [inline client/reader/recent/style.scss:3]: We import these styles in other DataViews implementations on Dotcom.

File: client/reader/recent/style.scss [hunk 3]
```diff
@@ -55,108 +57,90 @@
...
-			.dataviews-view-list__item-wrapper {
-				justify-content: start;
-
-				[role="gridcell"] {
-					width: 100%;
+			li {
+				border-top: 1px solid var( --studio-gray-0 );
+				
+				&.is-selected + li {
+					border-top: 1px solid var( --studio-gray-0 );
```
Comment (allilevine) [inline client/reader/recent/style.scss:71]: This is preventing a blue border under the selected item.

File: client/reader/recent/style.scss [hunk 4]
```diff
@@ -55,108 +57,90 @@
...
-			}
+				&.is-selected {
+					border-top: 1px solid #3858e940;
 
-			svg:hover {
-				fill: unset;
-			}
-		}
+					.dataviews-view-list__item-wrapper {
+						background-color: #f7faff;
```
Comment (allilevine) [inline client/reader/recent/style.scss:83]: This is changing the DataViews selected background color to the one in our design.

File: client/reader/recent/style.scss [hunk 5]
```diff
@@ -55,108 +57,90 @@
...
+				}
 
-			&__title {
-				min-width: 0;
-			}
+				.dataviews-view-list__media-wrapper {
+					border-radius: 50%;
+					width: 24px;
+					height: 24px;
+					background-color: transparent;
```
Comment (allilevine) [inline client/reader/recent/style.scss:95]: This style changes the shape of the DataViews' media field.

File: client/reader/recent/style.scss [hunk 6]
```diff
@@ -55,108 +57,90 @@
...
+						flex-shrink: 0;
+					}
+				}
 			}
 
-			&__featured-image {
-				max-width: 38px;
-				align-self: flex-end;
-				flex-shrink: 0;
+			.dataviews-view-list__item {
```
Comment (allilevine) [inline client/reader/recent/style.scss:142]: This style changes the padding of the DataViews items to match the design / maximize horizontal space.

**Review by eoigal — APPROVED** [review]
WFM


https://github.com/user-attachments/assets/b9623da6-65fd-4183-874f-084bb8f463df

**Pull request merged**